### PR TITLE
add forEach & variants, fixes #92

### LINKED ISSUE #92: side-effecting actions?
what do you think about adding side-effecting functions like:

```
List<T>.forEach: (T=>void) => void
Maybe.forEach(T=>void) => void
```

same for non empty list and so on. For ˙Maybe` we could name it `ifPresent` instead of `forEach`?

### PR DESCRIPTION
**Pull request by emmanueltouzery**

### CODE REVIEW

File: src/monet.js [hunk 1]
```diff
@@ -627,6 +633,16 @@
                 return fn(a) ? self : None()
             })
         },
+        forEach: function (fn) {
+            if (this.isSome()) {
```
Comment (ulfryk) [inline src/monet.js]: maybe use `.cata` internally ?

File: src/monet.js [hunk 2]
```diff
@@ -627,6 +633,16 @@
...
+                fn(this.val);
+            }
+        },
+        orElseRun: function (fn) {
+            if (!this.isSome()) {
+                fn();
```
Comment (ulfryk) [inline src/monet.js]: as above.. 

and in all other

File: test/either-spec.js [hunk 3]
```diff
@@ -92,6 +92,16 @@ describe('An Either', function () {
         it('renders as Right(value)', function() {
           expect(rightString.toString()).toBe('Right(abcd)')
         })
+        it('will execute side-effects on forEach', function() {
+          var t = []
+          rightString.forEach(function (x) {t.push(x)})
+          expect(t).toEqual(['abcd'])
```
Comment (ulfryk) [inline test/either-spec.js]: better use Jasmine spies here --> https://jasmine.github.io/2.4/introduction#section-Spies

and in all other side effects tests ;)

**Comment by emmanueltouzery** [comment]
just merged with the develop branch after the contains merge for now. will now deal with your feedback

File: src/monet.js [hunk 4]
```diff
@@ -931,6 +949,12 @@
         cata: function (leftFn, rightFn) {
             return this.isRightValue ? rightFn(this.value) : leftFn(this.value)
         },
+        forEach: function (fn) {
+            this.cata(function() {}, fn)
+        },
+        forEachLeft: function (fn) {
+            this.cata(fn, function() {})
+        },
```
Comment (ulfryk) [inline src/monet.js]: maybe you can create `var noop = function () {}` and reuse it in all places

**Comment by emmanueltouzery** [comment]
i think it's good now. Had to upgrade jasmine to get "toHaveBeenCalledTimes(0)".

**Pull request merged**